ui-v2/app/utils/get-object-pool.js

```diff
@@ -0,0 +1,52 @@
+export default function(dispose = function() {}, max, objects = []) {
+  return {
+    acquire: function(obj, id) {
+      // TODO: what should happen if an ID already exists
+      // should we ignore and release both? Or prevent from acquiring? Or generate a unique ID?
+      // what happens if we can't get an id via getId or .id?
+      // could potentially use Set
+      objects.push(obj);
+      if (typeof max !== 'undefined') {
```

Comment:
An object pool with no max set is just an array...why make setting a max optional?

Reply:
I suppose it's a collection of objects that get disposed of by a configurable function. That collection can be constrained to a certain length, or be an infinite length, the thing is the disposal.